### PR TITLE
cmake: detect `nettle` when building with GnuTLS

### PR DESCRIPTION
**Pull request by vszakats**
`nettle` is a direct dependency of curl, when building with GnuTLS.
Add a new `Find` module to detect it.

Also:
- GHA/macos: drop `nettle` hack no longer necessary.
- add `nettle` to `libcurl.pc`.
- also add `nettle` to `libcurl.pc` in autotools builds.

Follow-up to 781242ffa44a9f9b95b6da5ac5a1bf6372ec6257 #11967
Closes #14285

### CODE REVIEW

File: CMake/FindNettle.cmake [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+#
+###########################################################################
+# - Try to find the nettle library
+# Once done this will define
+#
+# NETTLE_FOUND - system has nettle
+# NETTLE_INCLUDE_DIRS - nettle include directories
+# NETTLE_LIBRARIES - nettle library names
+
+if(UNIX)
```
Comment (talregev) [inline CMake/FindNettle.cmake:31]: gnutls and nettle can found also on windows with pgk config

Comment (vszakats) [reply CMake/FindNettle.cmake:31]: This will be part of the PR pending, that's dealing with that. See the TODO added there.

**Comment by lazka** [comment]
I get this with 8.9.1:

```
  CMake Error at CMakeLists.txt:544 (find_package):
    By not providing "Findnettle.cmake" in CMAKE_MODULE_PATH this project has
    asked CMake to find a package configuration file provided by "nettle", but
    CMake did not find one.
  
    Could not find a package configuration file provided by "nettle" with any
    of the following names:
  
      nettleConfig.cmake
      nettle-config.cmake
  
    Add the installation prefix of "nettle" to CMAKE_PREFIX_PATH or set
    "nettle_DIR" to a directory containing one of the above files.  If "nettle"
    provides a separate development package or SDK, be sure it has been
    installed.
```

There is no Findnettle.cmake in the release tarball

**Comment by lazka** [comment]
I just took the file from git for now. With that all seems fine.

**Comment by vszakats** [comment]
Thanks for the report and I'm sorry about this. → #14320

**Comment by lazka** [comment]
No problem :) thanks!

**Pull request closed without merging**